lib/phoenix_live_view/plug.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Phoenix.LiveView.Plug do
+  @moduledoc """
+   Provides a Plug for LiveView rendering.
+
+   ## Options
+
+   See `Phoenix.LiveView.Router.live/3` for full set of supported options.
+
+   ## Examples
+
+       plug Phoenix.LiveView.Plug, MyApp.UserLive
+       plug Phoenix.LiveView.Plug, {MyApp.UserLive, session: [:path_params, :user_id]}
+
+  """
+
+  alias Phoenix.LiveView.Controller
+  alias Plug.Conn
+
+  @behaviour Plug
+
+  @impl Plug
+  def init(opts), do: opts
+
+  @impl Plug
+  @spec call(Conn.t(), module()) :: Conn.t()
+  def call(%Conn{} = conn, view) when is_atom(view) do
+    opts = Map.get(conn.private, :phoenix_live_view, [])
+    do_call(conn, view, opts)
+  end
+
+  @spec call(Conn.t(), {module(), Keyword.t()}) :: Conn.t()
+  def call(%Conn{} = conn, {view, opts}) when is_atom(view) and is_list(opts) do
+    do_call(conn, view, opts)
+  end
+
+  defp do_call(conn, view, opts) do
+    session_opts = Keyword.get(opts, :session, [:path_params])
+    render_opts = Keyword.merge(opts, session: session(conn, session_opts))
```

Comment:
I don't think we generally want to pass all of the options to render. I would do so explicitly. So here, I would do:

```suggestion
    render_opts = [session: session(conn, session_opts)]
```

Reply:
At least 1 other option must be forwarded. `:container` is a supported option now. How about
```
opts
|> Keyword.take([:container])
|> Keyword.put(:session, session(conn, session_opts))
```

Reply:
Beautiful, agreed!